engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -1124,39 +1302,62 @@ public long getNextSlaveSequenceValue( String slaveSequenceName, long incrementV
   }
 
   public SlaveServer getClient() {
-    String pHostName = getHostname();
-    String pPort = getPort();
-    String name = MessageFormat.format( "Dynamic slave [{0}:{1}]", pHostName, pPort );
-    SlaveServer client = new SlaveServer( name, pHostName, "" + pPort, getUsername(), getPassword() );
-    client.setSslMode( isSslMode() );
-    return client;
+    lock.readLock().lock();
+    try {
+      String pHostName = getHostname();
+      String pPort = getPort();
+      String name = MessageFormat.format( "Dynamic slave [{0}:{1}]", pHostName, pPort );
+      SlaveServer client = new SlaveServer( name, pHostName, "" + pPort, getUsername(), getPassword() );
```

Comment:
I believe the "" + pPort was a vestige of the old days when getPort() returned an integer (before it handled variables). I believe they just forgot to fix that notation. Since the class is being touched, it makes sense to fix it here.